libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/gqldirective.kt

```diff
@@ -122,26 +122,31 @@ private fun GQLValue?.toBoolean(): Boolean {
   }
 }
 
-private fun GQLValue?.toCatchTo(): CatchTo {
+private fun GQLValue?.toCatchTo(): CatchTo? {
   return when (this) {
     is GQLEnumValue -> when (this.value) {
       "NULL" -> CatchTo.NULL
       "RESULT" -> CatchTo.RESULT
       "THROW" -> CatchTo.THROW
-      else -> error("Unknown CatchTo value: ${this.value}")
+      else -> null
```

Comment:
For the record, we just discussed that and decided to move this to validation. Validation should check that the `@catch` directive definition is compatible with the one we're expecting, like for `@oneOf`